dandiapi/api/models/asset.py

```diff
@@ -86,7 +87,12 @@ class Status(models.TextChoices):
 
     asset_id = models.UUIDField(unique=True, default=uuid.uuid4)
     path = models.CharField(max_length=512)
-    blob = models.ForeignKey(AssetBlob, related_name='assets', on_delete=models.CASCADE)
+    blob = models.ForeignKey(
+        AssetBlob, related_name='assets', on_delete=models.CASCADE, null=True, blank=True
+    )
+    zarr = models.ForeignKey(
+        'ZarrArchive', related_name='assets', on_delete=models.CASCADE, null=True, blank=True
+    )
```

Comment:
is there some way on DB level to ensure that one of those (`blob` or `zarr`) is defined? if not -- we would be doomed to add some kind of a check to dashboard etc to ensure there are no "holes" right?

Reply:
https://github.com/dandi/dandi-api/pull/586/files#diff-62b3581f30735b317c48c250a4b709338e3beb99b09b8479ffda1a219c74b8e9R113-R120

Reply:
Coolio, thanks!